Apps/phproofpoint/proofpoint_connector.py

```diff
@@ -1,95 +1,239 @@
-PP_API_PATH_MESSAGES_DELIVERED = '/v2/siem/messages/delivered'
-PP_API_PATH_ISSUES = '/v2/siem/issues'
-PP_API_PATH_ALL = '/v2/siem/all'
-PP_API_PATH_CAMPAIGN = '/v2/campaign/{}'
-PP_API_PATH_FORENSICS = '/v2/forensics'
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
 
 
 class ProofpointConnector(BaseConnector):
     def __init__(self):
-        # Call the EmailConnector init first
+        # Call the BaseConnectors init first
         super(ProofpointConnector, self).__init__()
 
         self._pp_conn = None
         self._state_file_path = None
         self._state = {}
 
     def initialize(self):
+
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
         self._state = self.load_state()
+
+        # get the asset config
+        config = self.get_config()
+
+        self._username = config['username']
+        self._password = config['password']
+        self._headers = {
+            'X-Requested-With': 'REST API',
+            'Content-type': 'application/json',
+            'Accept': 'application/json'
+        }
+
+        # Fetching the Python major version
+        try:
+            self._python_version = int(sys.version_info[0])
+        except:
+            return self.set_status(phantom.APP_ERROR, ERROR_FETCHING_PYTHON_VERSION)
+
         return phantom.APP_SUCCESS
 
     def finalize(self):
         self.save_state(self._state)
         return phantom.APP_SUCCESS
 
-    def _make_rest_call(self, action_result, endpoint, params=None):
-        config = self.get_config()
-        user = config['username'].encode('utf-8')
-        password = config['password']
-        url = PP_API_BASE_URL + endpoint
-        if not params:
-            params = {}
-        params['format'] = 'json'
+    def _handle_py_ver_compat_for_input_str(self, input_str):
+        """
+        This method returns the encoded|original string based on the Python version.
+        :param input_str: Input string to be processed
+        :return: input_str (Processed input string based on following logic 'input_str - Python 3; encoded input_str - Python 2')
+        """
+
+        try:
+            if input_str and self._python_version == 2:
+                input_str = UnicodeDammit(input_str).unicode_markup.encode('utf-8')
+        except:
+            self.debug_print(PY_2TO3_ERROR_MSG)
+
+        return input_str
+
+    def _get_error_message_from_exception(self, e):
+        """ This method is used to get appropriate error message from the exception.
+        :param e: Exception object
+        :return: error message
+        """
+
+        try:
+            if e.args:
+                if len(e.args) > 1:
+                    error_code = e.args[0]
+                    error_msg = e.args[1]
+                elif len(e.args) == 1:
+                    error_code = ERROR_CODE_MSG
```

Comment:
Please change ERROR to ERR in all consts for consistency

Reply:
Fixed